frontend/model/contracts/group.js

```diff
@@ -1286,8 +1276,10 @@ sbp('chelonia/defineContract', {
       process ({ meta }, { state, getters }) {
         const period = getters.periodStampGivenDate(meta.createdDate)
         const current = getters.groupSettings?.distributionDate
-
-        if (current !== period) {
+        // Maybe we're updating the distribution date while in the waiting period.
+        if (!getters.groupDistributionStarted(meta.createdDate)) {
+          getters.groupSettings.distributionDate = meta.createdDate
+        } else if (period > current) {
```

Comment:
I'm pretty sure this change needs to be reverted and the code left unmodified.

This action is only called from a single place in the app: in `state.js` when we enter a new payment period and the distribution date needs to be moved forward.

`state.js`, in turn, never calls this when we're in the waiting period:

```js
          const distributionDateInSettings = store.getters.groupSettings.distributionDate
          const distributionStarted = store.getters.groupDistributionStarted(reactiveDate.date)
          if (oldPeriod && newPeriod && distributionStarted && (newPeriod !== distributionDateInSettings)) {
            sbp('gi.actions/group/updateDistributionDate', { contractID: store.state.currentGroupId })
          }
```

So this check doesn't make sense to add.

Finally, this sets the `distributionDate` to `meta.createdDate`. That also is incorrect, as the distribution date must be set to a periodstamp, not to any random date.

So I'm 99.9% certain this code needs to be reverted.

Reply:
Going to try reverting this, I think it was necessary at some point during the PR, but I'd rather get rid of it too